beacon-chain/state/state-native/getters_withdrawal.go

```diff
@@ -211,6 +174,55 @@ func (b *BeaconState) ExpectedWithdrawals() ([]*enginev1.Withdrawal, uint64, err
 	return withdrawals, processedPartialWithdrawalsCount, nil
 }
 
+func withdrawalTotal(ws []*enginev1.Withdrawal, idx primitives.ValidatorIndex) (total uint64) {
+	for _, w := range ws {
+		if w.ValidatorIndex == idx {
+			total += w.Amount
+		}
+	}
+	return
+}
+
+func (b *BeaconState) processPendingPartialWithdrawals() ([]*enginev1.Withdrawal, uint64, error) {
+	withdrawals := make([]*enginev1.Withdrawal, 0, params.BeaconConfig().MaxWithdrawalsPerPayload)
```

Comment:
This can just be `MaxPendingPartialsPerWithdrawalsSweep`, there's no reason to use `MaxWithdrawalsPerPayload` since it's own function now

```suggestion
	withdrawals := make([]*enginev1.Withdrawal, 0, params.BeaconConfig(). MaxPendingPartialsPerWithdrawalsSweep)
```

Reply:
but on line 124 I set withdrawals on the bigger function equal to this returned withdrawal wouldn't that override the capacity?

Reply:
or maybe I can do somethign so this one is partial withdrawals and copies into the withdrawals on line 124